tools/ScenarioTest.ResourceManager/PermissiveRecordMatcherWithApiExclusion.cs

```diff
@@ -116,22 +147,69 @@ protected bool ContainsIgnoredProvider(string requestUri, out string version)
                 !requestUri.Contains("/servicePrincipals?") &&
                 !requestUri.StartsWith("/webhdfs/v1/?aclspec", StringComparison.InvariantCultureIgnoreCase))
             {
-                version = String.Empty;
+                apiVersion = String.Empty;
                 return true;
             }
 
-            foreach (var provider in _providersToIgnore)
+            // Ignore resource providers
+            foreach (var provider in providersToIgnore)
             {
                 var providerString = string.Format("providers/{0}", provider.Key);
                 if (requestUri.Contains(providerString))
                 {
-                    version = provider.Value;
+                    apiVersion = provider.Value;
                     return true;
                 }
             }
 
+            if (resourcesToIgnore != null && resourcesToIgnore.Any())
+            {
+                // If we're looking at a specific provider and we have top level resource from this provider to ignore
+                foreach (var resourceToIgnore in resourcesToIgnore)
+                {
+                    var segments = requestUri.Split('/');
+
+                    // /subscriptions/.../resourceGroups/.../providers/Microsoft.X/resourceType...?api-version=Y
+                    if (requestUri.Contains("resourceGroups/") && requestUri.Contains("providers/"))
+                    {
+                        if (segments.Length > 8)
+                        {
+                            var resourceIdentifier = new ResourceIdentifier(requestUri);
+                            if (resourceIdentifier.ResourceType == resourceToIgnore)
+                            {
+                                apiVersion = String.Empty;
+                                return true;
+                            }
+                        }
+                        else if (segments.Length == 8)
+                        {
+                            var resourceType = $"{segments[6]}/{segments[7]}";
+                            if (resourceType.Contains(resourceToIgnore))
+                            {
+                                apiVersion = String.Empty;
+                                return true;
+                            }
+                        }
+                    }
+
+                    // /subscriptions/.../providers/Microsoft.Provider/resourceType
+                    if (requestUri.Contains("providers/"))
```

Comment:
@johnpaulkee same comment here where we should use regex:

```cs
var regex = new Regex(@"\/subscriptions\/[0-9A-Fa-f-]*\/providers\/[A-Za-z]*.[A-Za-z]*\/.*?api-version=[0-9-]*");
```

Reply:
Great idea, will update in pool PR https://github.com/Azure/azure-powershell/pull/9744

Reply:
Updated the regex slightly.
@"\/subscriptions\/[0-9A-Fa-f-]*\/providers\/[A-Za-z]*.[A-Za-z]*\/.*?**[api-version=[0-9-]*]?**"